torch/distributed/fsdp/shard_utils.py

```diff
@@ -188,3 +194,56 @@ def _gather_state_dict(
             tensor = output_tensor
         new_state_dict[key] = tensor
     return new_state_dict
+
+
+def _create_chunk_sharded_tensor(
+    tensor: torch.Tensor,
+    rank: int,
+    world_size: int,
+    device_per_node: int,
+    pg: dist.ProcessGroup,
+) -> ShardedTensor:
+    """
+    Shard a tensor to chunks along the first dimension. The local rank will gets its
+    corresponding chunk as the local shard to create a ShardedTensor.
+    """
+    chunks = tensor.chunk(world_size)
+    if len(chunks) > rank:
+        local_shard = chunks[rank].clone()
+        offsets = [0 for _ in tensor.size()]
+        offsets[0] = math.ceil(tensor.size()[0] / world_size) * rank
+        local_shards = [Shard.from_tensor_and_offsets(local_shard, offsets, rank)]
```

Comment:
Conceptual check: `local_shards` here has length 1? (So overall, it is either length 1 if this rank has a chunk, or length 0 if this rank does not have a chunk?)

Reply:
That's correct!